src/rules/dto-decorators.ts

```diff
@@ -0,0 +1,221 @@
+          );
+
+          const classValidatorDecs = decorators
+            .filter(
+              dec =>
+                classValidatorDecorators.has(dec.expression.callee.name) &&
+                dec.expression.callee.name !== "IsOptional",
+            )
+            .sort((a, b) => {
+              if (b.expression.callee.name > a.expression.callee.name) {
+                return -1;
+              } else return 1;
+            });
+
+          const optionalityDecs = decorators.filter(dec =>
+            optionalDecorator.has(dec.expression.callee.name),
+          );
+
+          const customDecs = decorators
+            .filter(
+              dec => !allNonCustomDecorators.has(dec.expression.callee.name),
+            )
+            .sort((a, b) => {
+              if (b.expression.callee.name > a.expression.callee.name) {
+                return -1;
+              } else return 1;
+            });
+
+          const orderedDecorators = [
+            ...swaggerDecs,
+            ...classTransformerDecs,
+            ...classValidatorDecs,
+            ...optionalityDecs,
+            ...customDecs,
+          ];
+
+          const firstImportNode = decorators[0];
+          const lastImportNode = decorators[decorators.length - 1];
+
+          return {
+            decorators: orderedDecorators,
+            text: orderedDecorators
+              .map(node => {
+                return sourceCode.getText(node as unknown as Node).trim();
+              })
+              .join("\n")
+              .trim(),
+            start: firstImportNode.range ? firstImportNode.range[0] : 0,
+            end: lastImportNode.range ? lastImportNode.range[1] : 0,
+          };
+        }
+
+        if (
+          swaggerDecorators.size === 0 &&
+          classValidatorDecorators.size === 0 &&
+          classTransformerDecorators.size === 0
+        ) {
+          return;
+        }
+
+        const sourceCode = context.getSourceCode();
+
+        const classNode = body.find(
+          el =>
+            el.type === "ClassDeclaration" ||
+            ((el.type === "ExportNamedDeclaration" ||
+              el.type === "ExportDefaultDeclaration") &&
+              el.declaration?.type === "ClassDeclaration"),
+        ) as ClassDeclaration | undefined;
+
+        if (!classNode) {
+          return;
+        }
+
+        let classBody;
+
+        // @ts-ignore
+        if (!classNode.body && !classNode.declaration) {
+          return;
+        } else if (classNode.body) {
+          if (classNode.body.body) {
+            classBody = classNode.body.body;
+          } else {
+            return;
+          }
+        } else {
+          if (
+            // @ts-ignore
+            !classNode.declaration.body &&
+            // @ts-ignore
+            !classNode.declaration.body.body
+          ) {
+            return;
+          } else {
+            // @ts-ignore
+            classBody = classNode.declaration.body.body;
+          }
+        }
```

Comment:
Would this be enough?
```ts
const classBody = classNode.body?.body || 
  // @ts-expect-error
  classNode.declaration?.body?.body
```
Also, it's better to use `@ts-expect-error` since the directive itself will show an error once the underlying code can no longer throw any errors.

Reply:
Yes, using the conditional chaining can do the same thing.